lib/generics.go

```diff
@@ -55,3 +55,19 @@ func (set *Set[T]) ToSlice() []T {
 	}
 	return results
 }
+
+func MapSet[T comparable, K any](set *Set[T], mapFunc func(elem T) (K, error)) ([]K, error) {
+	var results []K
+	err := set.ForEach(func(elem T) error {
+		mappedResult, innerErr := mapFunc(elem)
+		if innerErr != nil {
+			return innerErr
```

Comment:
I assume this works, but I'm not sure I follow how. We will end up with a mappedResult that is less than or equal to the length of the set. Less than if there are any errors. Equal if there aren't any errors. And we will end up with a map of returned errors or nils. How does the outer error know to error if e.g. `Any?([nil, nil, error, nil, nil])`.

Reply:
If any element returns an error from the map function, the whole thing errors. I can add a test for this

Reply:
Makes sense. +1 to a test. Would convince me definitively.